inference-engine/src/multi_device/multi_device_plugin.cpp

```diff
@@ -169,20 +170,30 @@ ExecutableNetworkInternal::Ptr MultiDeviceInferencePlugin::LoadExeNetworkImpl(co
     std::unordered_map<std::string, InferenceEngine::Parameter> multiNetworkConfig;
     multiNetworkConfig.insert(*priorities);
 
+    bool enablePerfCounters = false;
     DeviceMap<ExecutableNetwork> executableNetworkPerDevice;
+    std::mutex load_mutex;
+    std::vector<Task> loads;
     for (auto& p : metaDevices) {
-        auto & deviceName = p.deviceName;
-        auto & deviceConfig = p.config;
-        executableNetworkPerDevice.insert({ deviceName, GetCore()->LoadNetwork(network, deviceName, deviceConfig) });
-        multiNetworkConfig.insert(deviceConfig.begin(), deviceConfig.end());
-    }
+        loads.push_back([&]() {
+            const auto &deviceName = p.deviceName;
+            const auto &deviceConfig = p.config;
+            auto exec_net = GetCore()->LoadNetwork(network, deviceName, deviceConfig);
+            // getting the config from the loaded network to respect both device's plugin and load-specific setting
+            enablePerfCounters |= exec_net.GetConfig(PluginConfigParams::KEY_PERF_COUNT).as<std::string>() == PluginConfigParams::YES;
```

Comment:
not sure that all plugins returns this correctly

Reply:
good point, fixed